pkg/routing/adapter/filter/adapter_test.go

```diff
@@ -0,0 +1,301 @@
+/*
+Copyright 2022 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filter
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/client-go/rest"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+
+	"knative.dev/eventing/pkg/kncloudevents"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/injection"
+	logtesting "knative.dev/pkg/logging/testing"
+	rectesting "knative.dev/pkg/reconciler/testing"
+
+	common "github.com/triggermesh/triggermesh/pkg/apis/common/v1alpha1"
+	v1alpha1 "github.com/triggermesh/triggermesh/pkg/apis/routing/v1alpha1"
+	fakeinjectionclient "github.com/triggermesh/triggermesh/pkg/client/generated/injection/client/fake"
+	fakeinformer "github.com/triggermesh/triggermesh/pkg/client/generated/injection/informers/routing/v1alpha1/filter/fake"
+)
+
+const (
+	testServerPort = 8989
```

Comment:
It would be less error prone if we used a random port in each test.

Here is a little trick I stole from the standard Go library: https://github.com/triggermesh/mux/blob/9dd9f26ab0/pkg/routing/adapter/mux/adapter_test.go#L62-L64